cf/data/fragment/mixin/fragmentarraymixin.py

```diff
@@ -244,33 +166,12 @@ def _size_1_axis(self, indices):
         None
 
         """
-        axis = None
-
-        n_size_1 = 0  # Number of size 1 indices
-        for i, (index, n) in enumerate(zip(indices, self.shape)):
-            try:
-                x = index.indices(n)
-                if abs(x[1] - x[0]) == 1:
-                    # Index is a size 1 slice
-                    n_size_1 += 1
-                    axis = i
-            except AttributeError:
-                try:
-                    if index.size == 1:
-                        # Index is a size 1 numpy or dask array
-                        n_size_1 += 1
-                        axis = i
-                except AttributeError:
-                    if len(index) == 1:
-                        # Index is a size 1 list
-                        n_size_1 += 1
-                        axis = i
-
-        if n_size_1 > 1:
-            # There are two or more size 1 indices
-            axis = None
-
-        return axis
+        # REVIEW: getitem: `_size_1_axis`: refactor to use `original_shape`
+        original_shape = self.original_shape
```

Comment:
Hi Sadie - took me while to work out what going on here ... `NetCDF4FragmentArray` inherits from `(FragmentArrayMixin, NetCDF4Array)`, and  `NetCDF4Array` inherits from `IndexMixin`, thereby giving us our `def original_shape` definition.

Reply:
Ah OK yes I see now after studying the diagram a bit more closely (lots of curvy lines to trace back and forth :slightly_smiling_face: ), so it all works out so that all of the relevant `*Array` classes (all those except `FileArray` and `Array` it looks like) inherit from `IndexMixin`. Thanks for clarifying. All good, marking as resolved.